tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java

```diff
@@ -1143,7 +1144,12 @@ public synchronized void voteForValue(User user, String distinguishingXpath, Str
                 }
             }
 
+            String oldVal = xmlsource.getValueAtDPath(distinguishingXpath);
```

Comment:
looks like this is unused except when newVal != null, so put under that condition.

Reply:
OK -- sorry I merged before I saw your comment. I'll do that in another PR.

Reply:
On further examination, we need to get oldVal before processing the vote, and we don't know whether newVal will be null until after processing the vote, so we need to get oldVal unconditionally. Note that newVal is the winning value after the vote, and isn't necessarily the same as "value", which is what the user voted for. "value" is null if the user abstained, but newVal is only null if there's no winning value, not even an inherited or baseline value. This function is only called once, when a user submits a vote, so it's not really a hot spot or inner-loop context. OK?